.github/pull_request_template.md

```diff
@@ -0,0 +1,3 @@
+**Checklist**
+- [ ] Tests
+- [ ] Update developer documentation
```

Comment:
Maybe split it up into 
- Update docstrings.
- Update user manual.

I know that I'll help me !

Also do we need : 
- Update type annotations (optional)?

Reply:
> Maybe split it up into
>
> Update docstrings.
> Update user manual.
> I know that I'll help me !

Good point. Thank you!
Maybe I should say "API reference" (`docs/source/traits_api_reference`)  instead of "developer documentation" or "docstring"?
It is expected that docstring will be updated with the code, because they are co-located. The sphinx doc are the ones that tend to get forgotten.

> Also do we need :
> Update type annotations (optional)?

For my curiosity: What does one need to do for this task?

Anyone can open a PR on traits. It would be good to keep the checklist simple so we don't turn away first-time contributors (or any contributors really).  If it is optional, can we skip this for now?

Reply:
This is for [EEP-4](https://github.com/enthought/ets/pull/29) for which an initial attempt has been merged into master Ref: [here](https://github.com/enthought/traits/pull/904)

Updating type annotations is only required if a trait type has been added/modified.

Reply:
Thanks. I added that now.